### DISCUSSION TITLE
Define a model property name as dynamic

### DISCUSSION THREAD

**Discussion by myakimov** [microsoft/typespec#1626] [Q&A] [answered]
Hello,

I am trying to define a model with a property which will be with dynamic name (unknown at design time), technically the property I want to define can take any valid identfier. In OpenAPI I can represent this using the `propertyNames` and `patternProperties`. Is there a way for me to define such a property in Cadl definition ? 

Thanks in advance!

**Comment by markcowl** [comment, marked as answer]
Hmmm, so it is possible to add additional undefined properties by using `Record<T>`  but that would allow any number of properties with unknown keys.

For example:

```tsp
model Foo is Record<unknown> {}
```

Which would allow properties with valid keys and any type.

**Comment by laduke** [comment]
Is there any way to specify anything about the keys?

I'd like to put something like `@pattern("[a-f0-9]{10}")` on the keys of something like

`alias MyAnnoyingType = Record<int32>;`

It's unfortunately a Map of <IdString, int32>

When I emit to openapi, I get:

```
      responses:
        '200':
          description: The request has succeeded.
          content:
            application/json:
              schema:
                type: object
                additionalProperties:
                  type: integer
                  format: int32
```

and then docs generators don't have much to go on and render something like:

```
{
  "property1": 0,
  "property2": 0
}
```

**Reply by timotheeguerin** [reply]
it is unfortunately not possible today, this was a bit of a mistake when designing `Record<T>` could you file an new issue
